TheWire rework UI, and update Data Messages.

drbob opened the pull request:
Significant rework of The Wire to support:
 * A interface closer to Twitter.
 * Replies / Republishes / Likes
 * Up to 4 Images

drbob committed:
update libretroshare interface for TheWire

expanded groups and msgs with additional data and images.
new interfaces for improved ui display of message
start moving fetch logic to libretroshare with blocking APIs.

drbob committed:
Improvements to UI to make behaviour more similar to twitter.

Add new UI elements.
Can show PulseFocus or GroupFocus at the moment.
WIP, Much more work to get this finalised.

drbob committed:
Move createOriginalPulse / createReplyPulse to libretroshare

message creation logic moved to libretroshare
GUI classes cleaned up.

drbob committed:
Enable Republish / Likes from new UI elements

Update interface to use groupId / msgId.
Tweak PulseAddDialog for Republish & like.
Hide Like messages from showing in UI.
Wire up buttons from new UI elements.

drbob committed:
fixup button labels are prefix headers

drbob committed:
Add image handling to PulseAddDialog and PulseMessage

drbob committed:
Fixup how REF message are displayed

Ensure RefImageCount is correct.
Update docs on m(Ref)GroupPtr points to
fill-in GroupPtrs on Ref Msgs.
Remove mRefAuthorName as not so useful.
Add images into Response Msgs.
Display Headshot / Images on Ref Msgs - if possible.

defnax commented:
i like this design lots better then the other one :)

csoler commented on libretroshare/src/services/p3wire.cc:
```diff
@@ -31,6 +31,23 @@
 
+uint32_t RsWirePulse::ImageCount()
+{
+	if (!mImage4.empty()) {
```
The logic seems risky here. One can craft a msg with mImage3=nullptr and mImage4 not empty. But still the number of images will not be 4.

drbob replied:
made this more robust

drbob committed:
use Extra Widget for Group creation

drbob committed:
Update display of WireGroup

defnax commented:
at the moment i doesnt like the Show Group & Show Parent Buttons which not needed there
best we need a Back Button on the same place like on twitter on top left
![image](https://user-images.githubusercontent.com/9952056/83430759-808af100-a436-11ea-8216-a6240ed2bbc6.png)

@drbob i have a little pr changes for the group item
https://github.com/drbob/RetroShare/pull/1

![image](https://user-images.githubusercontent.com/9952056/83058683-0b47a680-a059-11ea-9d62-b802816f8d38.png)

defnax committed:
Added to display the headshot images on the group item

drbob committed:
Merge branch 'master' into TheWire-rework-ui

drbob committed:
Merge branch 'master' into TheWire-rework-ui

drbob committed:
Merge remote-tracking branch 'origin/master' into TheWire-rework-ui

drbob committed:
Remove old Pulse display

Remove PulseItem / PulseDetails classes
Switch all Message retrieval away from TokenQueues
Remove Original PulseItem View

drbob committed:
Merge pull request #1 from defnax/wire-changes

Added to display the headshot images on the group item

drbob commented:
Defnax: I like the idea of a back button + history, I'll try and add this in the next PR.
I think the "Show Parent" and "Show Group" are necessary - just to allow fluid navigation.
Obviously, I'll change the buttons into clickable Labels, in the future.
PS. I've merged your PR.

drbob committed:
make image count more robust
